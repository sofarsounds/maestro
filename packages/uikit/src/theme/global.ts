import { createGlobalStyle, css } from '../lib/styledComponents';

import OpenSansRegular from '../assets/fonts/OpenSans-Regular.ttf';
import OpenSansSemiBold from '../assets/fonts/OpenSans-SemiBold.ttf';
import OpenSansBold from '../assets/fonts/OpenSans-Bold.ttf';
import OpenSansExtraBold from '../assets/fonts/OpenSans-ExtraBold.ttf';

import IconFontTTF from '../assets/fonts/sofarsounds-icon-font.ttf';
import IconFontWOFF from '../assets/fonts/sofarsounds-icon-font.woff';
import IconFontSVG from '../assets/fonts/sofarsounds-icon-font.svg';

export default createGlobalStyle`
  ${({ theme }) => css`
    @font-face {
      font-family: 'sofarsounds-icon-font';
      src:
        url(${IconFontTTF}) format('truetype'),
        url(${IconFontWOFF}) format('woff'),
        url(${IconFontSVG}) format('svg');
      font-weight: normal;
      font-style: normal;
    }

    @font-face {
      font-family: 'Open Sans Regular';
      src:
        url(${OpenSansRegular}) format('truetype'),
        url(${OpenSansSemiBold}) format('truetype'),
        url(${OpenSansBold}) format('truetype'),
        url(${OpenSansExtraBold}) format('truetype');
      font-style: normal;
    }

    body,
    html,
    * {
      font-family: "${theme.fonts.regular}", sans-serif;
      color: ${theme.colours.backToBlack};
    }

    body {
      font-size: ${theme.fontSizes.base};
    }

    html {
      box-sizing: border-box;
    }

    *,
    ::after,
    ::before {
      box-sizing: inherit;
    }

  `}
`;
