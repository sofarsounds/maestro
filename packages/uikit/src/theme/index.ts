import { css } from '../lib/styledComponents';

import ITheme from '../typings/theme';

const breakPoints = {
  phone: 0,
  tablet: 768,
  desktop: 992
};

const theme: ITheme = {
  colours: {
    primary: '#008736',

    // blacks & greys
    black: '#000000',
    backToBlack: '#111111',
    blackBetty: '#1E1E1E',
    paintItBlack: '#333333',
    blueSmoke: '#949494',
    macyGrey: '#DBDBDB',
    silverSprings: '#F5F5F5',
    whiteDenim: '#FFFFFF',

    // green variations
    green900: '#006721',
    green700: '#039842',
    green600: '#10AD52',
    green500: '#23BA57',
    green400: '#4FC571',
    green300: '#72D08A',
    green200: '#9DDCAC',
    green100: '#CFEACC',
    green50: '#E6F7EA',

    // additional
    redRedWine: '#EF4423'
  },

  fonts: {
    fontFamily: 'Open Sans'
  },
  fontSizes: {
    base: '16px',
    h1: '96px',
    h2: '60px',
    h3: '48px',
    h4: '34px',
    h5: '24px',
    h6: '20px',
    title: '74px',
    subtitle1: '16px',
    subtitle2: '14px',
    body1: '16px',
    body2: '14px',
    overline: '10px',
    caption: '10px',
    button: '14px'
  },

  media: Object.keys(breakPoints).reduce((acc, label: string) => {
    acc[label] = (...args: any) => css`
      @media (min-width: ${breakPoints[label]}px) {
        ${css.call(undefined, ...args)};
      }
    `;
    return acc;
  }, {}),

  borderRadius: {
    button: '4px'
  },

  utils: {
    transition: (attr?: string) => `${attr || 'all'} .2s ease`
  }
};

export default theme;
