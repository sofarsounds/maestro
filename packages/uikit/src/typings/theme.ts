/**
 * Main Theme File
 * This is where all variables and colours are defined that can
 * be accessed using props in every component
 *
 * @TODO the colours collection on the zeplin guide is ridiculous so I left out any
 * colours that weren't used.
 *
 * @TODO fontSizes.. Are really ALL off the different headings being used? 2 types for body and
 * subtitle? We should maybe have a look at making this a bit easier.
 */

export default interface Theme {
  colours: {
    primary: string;

    // blacks & greys
    black: string;
    backToBlack: string;
    blackBetty: string;
    paintItBlack: string;
    blueSmoke: string;
    macyGrey: string;
    silverSprings: string;
    whiteDenim: string;

    // green variations
    green900: string;
    green800: string;
    green700: string;
    green600: string;
    green500: string;
    green400: string;
    green300: string;
    green200: string;
    green100: string;
    green50: string;

    // additional
    redRedWine: string;
  };

  fonts: {
    regular: string;
  };
  fontSizes: {
    base: string;
    h1: string;
    h2: string;
    h3: string;
    h4: string;
    h5: string;
    h6: string;
    title: string;
    subtitle1: string;
    subtitle2: string;
    body1: string;
    body2: string;
    overline: string;
    caption: string;
    button: string;
    formGroupLabel: string;
  };

  media: any;

  ruler: number[];
  zIndex: {
    navbar: number;
  };

  dimensions: {
    inputHeight: string;
    buttonHeight: string;
    containerWidth: {
      small: string;
      default: string;
      large: string;
      fullscreen: string;
    };

    button: {
      small: {
        height: string;
        padding: string;
      };

      default: {
        height: string;
        padding: string;
      };
    };
    navbarHeight: {
      lg: string;
      md: string;
      xs: string;
    };
  };

  borderRadius: {
    default: string;
    button: string;
    input: string;
  };
  utils: {
    transition: (a?: string) => string;
  };
}
