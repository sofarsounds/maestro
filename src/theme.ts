import Color from 'color';

export default {
  colors: {
    primary: '#3498db',
    secondary: '#8e44ad',
    white: '#ffffff'
  },
  utils: {
    borderRadius: '4px',
    transition: () => `all .25s ease`,
    darken: (hex: string, factor: number = 1) =>
      Color(hex)
        .darken(0.2 * factor)
        .hex()
  }
};
