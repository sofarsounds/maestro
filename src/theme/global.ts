import { createGlobalStyle, css } from '../lib/styledComponents';

import IconFontTTF from '../assets/fonts/sofarsounds-icon-font.ttf';
import IconFontWOFF from '../assets/fonts/sofarsounds-icon-font.woff';
import IconFontSVG from '../assets/fonts/sofarsounds-icon-font.svg';

export default createGlobalStyle`
  ${({ theme }) => css`
    /* open-sans-regular - latin */
    @font-face {
      font-family: 'Open Sans';
      font-style: normal;
      font-weight: 400;
      src: url('../assets/fonts/open-sans-v26-latin-regular.eot'); /* IE9 Compat Modes */
      src: local(''),
           url('../assets/fonts/open-sans-v26-latin-regular.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
           url('../assets/fonts/open-sans-v26-latin-regular.woff2') format('woff2'), /* Super Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-regular.woff') format('woff'), /* Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-regular.ttf') format('truetype'), /* Safari, Android, iOS */
           url('../assets/fonts/open-sans-v26-latin-regular.svg#OpenSans') format('svg'); /* Legacy iOS */
    }
    /* open-sans-600 - latin */
    @font-face {
      font-family: 'Open Sans';
      font-style: normal;
      font-weight: 600;
      src: url('../assets/fonts/open-sans-v26-latin-600.eot'); /* IE9 Compat Modes */
      src: local(''),
           url('../assets/fonts/open-sans-v26-latin-600.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
           url('../assets/fonts/open-sans-v26-latin-600.woff2') format('woff2'), /* Super Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-600.woff') format('woff'), /* Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-600.ttf') format('truetype'), /* Safari, Android, iOS */
           url('../assets/fonts/open-sans-v26-latin-600.svg#OpenSans') format('svg'); /* Legacy iOS */
    }
    /* open-sans-700 - latin */
    @font-face {
      font-family: 'Open Sans';
      font-style: normal;
      font-weight: 700;
      src: url('../assets/fonts/open-sans-v26-latin-700.eot'); /* IE9 Compat Modes */
      src: local(''),
           url('../assets/fonts/open-sans-v26-latin-700.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
           url('../assets/fonts/open-sans-v26-latin-700.woff2') format('woff2'), /* Super Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-700.woff') format('woff'), /* Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-700.ttf') format('truetype'), /* Safari, Android, iOS */
           url('../assets/fonts/open-sans-v26-latin-700.svg#OpenSans') format('svg'); /* Legacy iOS */
    }
    /* open-sans-800 - latin */
    @font-face {
      font-family: 'Open Sans';
      font-style: normal;
      font-weight: 800;
      src: url('../assets/fonts/open-sans-v26-latin-800.eot'); /* IE9 Compat Modes */
      src: local(''),
           url('../assets/fonts/open-sans-v26-latin-800.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
           url('../assets/fonts/open-sans-v26-latin-800.woff2') format('woff2'), /* Super Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-800.woff') format('woff'), /* Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-800.ttf') format('truetype'), /* Safari, Android, iOS */
           url('../assets/fonts/open-sans-v26-latin-800.svg#OpenSans') format('svg'); /* Legacy iOS */
    }

    @font-face {
      font-family: 'sofarsounds-icon-font';
      src:
        url(${IconFontTTF}) format('truetype'),
        url(${IconFontWOFF}) format('woff'),
        url(${IconFontSVG}) format('svg');
      font-weight: normal;
      font-style: normal;
    }

    body,
    html,
    * {
      font-family: "${theme.fonts.regular}", sans-serif;
    }

    body {
      font-size: ${theme.fontSizes.base};
      color: ${theme.colors.backToBlack};
      margin: 0px;
    }

    html {
      box-sizing: border-box;
    }

    *,
    ::after,
    ::before {
      box-sizing: inherit;
    }

    a {
      color: ${theme.colors.backToBlack};
    }

    a:link, a:visited {
      text-decoration: none;
    }

    button {
      background: none;
      margin-left: 0px;
      margin-right: 0px;
    }
  `}
`;
