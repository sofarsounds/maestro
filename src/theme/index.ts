import { css } from '../lib/styledComponents';

import Theme from '../typings/theme';

// breakpoints define the minimum width of a device (>=)
export const breakPoints = {
  xs: 0,
  sm: 576,
  md: 768,
  lg: 992,
  xl: 1200
};

const theme: Theme = {
  colors: {
    primary: '#039842',

    // blacks & greys
    black: '#000000',
    backToBlack: '#111111',
    blackBetty: '#1E1E1E',
    paintItBlack: '#333333',
    blueSmoke: '#949494',
    macyGrey: '#DBDBDB',
    silverSprings: '#F5F5F5',
    whiteDenim: '#FFFFFF',

    // green variations
    green900: '#006721',
    green800: '#008736',
    green700: '#039842',
    green600: '#10AD52',
    green500: '#23BA57',
    green400: '#4FC571',
    green300: '#72D08A',
    green200: '#9DDCAC',
    green100: '#CFEACC',
    green50: '#E6F7EA',

    // additional
    redRedWine: '#EF4423'
  },

  fonts: {
    regular: 'Open Sans'
  },
  fontSizes: {
    base: '16px',
    title: { lg: '54px', md: '38px', xs: '26px' },
    h1: { lg: '48px', md: '34px', xs: '24px' },
    h2: { lg: '34px', md: '24px', xs: '20px' },
    h3: { lg: '24px', md: '20px', xs: '16px' },
    h4: { lg: '20px', md: '16px', xs: '16px' },
    h5: { lg: '16px', md: '14px', xs: '14px' },
    h6: { lg: '14px', md: '14px', xs: '14px' },
    body1: '16px',
    body2: '14px',
    overline: '10px',
    caption: '10px',
    button: '14px',
    formGroupLabel: '14px'
  },

  media: Object.keys(breakPoints).reduce((acc, label: string) => {
    acc[label] = (...args: any) => css`
      @media (min-width: ${breakPoints[label]}px) {
        ${css.call(undefined, ...args)};
      }
    `;
    return acc;
  }, {}),

  ruler: Array.from({ length: 21 }, (e, i) => i * 4),
  zIndex: {
    navbar: 100
  },

  dimensions: {
    inputHeight: '44px',
    buttonHeight: '44px',
    radioBox: '20px',
    containerWidth: {
      small: '600px',
      default: '800px',
      large: '1000px',
      fullscreen: '100%'
    },

    button: {
      small: {
        height: '40px',
        padding: '0 15px'
      },

      default: {
        height: '44px',
        padding: '0 25px'
      }
    },

    navbarHeight: {
      lg: '84px',
      md: '74px',
      xs: '62px'
    }
  },

  borderRadius: {
    default: '2px',
    large: '8px',
    button: '4px',
    input: '2px'
  },

  utils: {
    transition: (attr?: string) => `${attr || 'all'} .2s ease`
  }
};

export default theme;
