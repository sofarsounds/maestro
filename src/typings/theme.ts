/**
 * Main Theme File
 * This is where all variables and colors are defined that can
 * be accessed using props in every component
 */

interface FontSizeResponsive {
  xs: string;
  md: string;
  lg: string;
}

export type Colors =
  | 'primary'
  | 'black'
  | 'blackBetty'
  | 'paintItBlack'
  | 'darkSmoke'
  | 'blueSmoke'
  | 'lightBlueSmoke'
  | 'macyGrey'
  | 'darkGrey'
  | 'silverSprings'
  | 'whiteDenim'
  | 'green900'
  | 'green800'
  | 'green700'
  | 'green600'
  | 'green500'
  | 'green400'
  | 'green300'
  | 'green200'
  | 'green100'
  | 'green50'
  | 'redRedWine'
  | 'greenOnions'
  | 'greenGarden'
  | 'blueChristmas'
  | 'purpleHaze'
  | 'purplePills'
  | 'checkBerry'
  | 'blueMagic'
  | 'brownSugar'
  | 'pinkTriangle'
  | 'royalBlood'
  | 'peaches'
  | 'orangeCrush'
  | 'mellowYellow'
  | 'yellowSubmarine';

export default interface Theme {
  colors: {
    primary: string;

    // blacks & greys
    black: string;
    backToBlack: string;
    blackBetty: string;
    paintItBlack: string;
    darkSmoke: string;
    blueSmoke: string;
    lightBlueSmoke: string;
    macyGrey: string;
    darkGrey: string;
    silverSprings: string;
    whiteDenim: string;

    // green variations
    green900: string;
    green800: string;
    green700: string;
    green600: string;
    green500: string;
    green400: string;
    green300: string;
    green200: string;
    green100: string;
    green50: string;

    // additional
    greenOnions: string;
    greenGarden: string;
    purpleHaze: string;
    blueChristmas: string;
    purplePills: string;
    checkBerry: string;
    blueMagic: string;
    brownSugar: string;
    pinkTriangle: string;
    royalBlood: string;
    redRedWine: string;
    peaches: string;
    orangeCrush: string;
    mellowYellow: string;
    yellowSubmarine: string;
  };

  fonts: {
    regular: string;
  };
  fontSizes: {
    base: string;
    h1: FontSizeResponsive;
    h2: FontSizeResponsive;
    h3: FontSizeResponsive;
    h4: FontSizeResponsive;
    h5: FontSizeResponsive;
    h6: FontSizeResponsive;
    title: FontSizeResponsive;
    body1: string;
    body2: string;
    overline: string;
    caption: string;
    button: string;
    formGroupLabel: string;
  };

  media: any;

  ruler: number[];
  zIndex: {
    navbar: number;
  };

  dimensions: {
    inputHeight: string;
    buttonHeight: string;
    radioBox: string;
    containerWidth: {
      small: string;
      default: string;
      large: string;
      fullscreen: string;
    };

    button: {
      small: {
        height: string;
        padding: string;
      };

      default: {
        height: string;
        padding: string;
      };
    };
    navbarHeight: {
      lg: string;
      md: string;
      xs: string;
    };
  };

  borderRadius: {
    default: string;
    large: string;
    button: string;
    input: string;
  };
  utils: {
    transition: (a?: string) => string;
  };
}
