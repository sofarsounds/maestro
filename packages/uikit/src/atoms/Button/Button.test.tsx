import React from 'react';
import { mountWithTheme } from '../../test';
import theme from '../../theme';
import { BaseButton, PrimaryButton, OutlineButton, LinkButton } from './index';

const onClickMock = jest.fn();

const setup = (
  Component: any,
  disabled: boolean = false,
  loading: boolean = false,
  colour: string = 'primary'
) =>
  mountWithTheme(
    <Component
      disabled={disabled}
      loading={loading}
      colour={colour}
      onClick={() => onClickMock()}
    >
      Button
    </Component>
  );

describe('Buttons', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Base Button', () => {
    it('is clickable', () => {
      const wrapper = setup(BaseButton, false, false);
      wrapper.simulate('click');
      expect(onClickMock).toHaveBeenCalled();
    });

    it('is not clickable in disabled state', () => {
      const wrapper = setup(BaseButton, true, false);
      expect(wrapper.find(BaseButton).prop('disabled')).toBeTruthy();
      wrapper.simulate('click');
      expect(onClickMock).not.toHaveBeenCalled();
    });

    it('is not clickable in loading state', () => {
      const wrapper = setup(BaseButton, true, true);
      expect(wrapper.find(BaseButton).prop('loading')).toBeTruthy();
      wrapper.simulate('click');
      expect(onClickMock).not.toHaveBeenCalled();
    });
  });

  describe('Primary Button', () => {
    it('renders correctly', () => {
      expect(setup(PrimaryButton)).toMatchSnapshot();
    });

    it('renders correctly in disabled state', () => {
      expect(setup(PrimaryButton, true)).toMatchSnapshot();
    });

    it('renders correctly in loading state', () => {
      expect(setup(PrimaryButton, false, true)).toMatchSnapshot();
    });
  });

  describe('Outline Button', () => {
    it('renders correctly', () => {
      expect(setup(OutlineButton)).toMatchSnapshot();
    });

    it('renders white outline Button correctly', () => {
      const wrapper = setup(OutlineButton, false, false, 'primary');
      expect(wrapper.find(OutlineButton).prop('colour')).toBe('primary');
      expect(wrapper).toHaveStyleRule('color', theme.colours.primary);
      expect(wrapper).toHaveStyleRule('background', theme.colours.whiteDenim);
      expect(wrapper).toHaveStyleRule(
        'border',
        `1px solid ${theme.colours.primary}`
      );
    });

    it('renders correctly in disabled state', () => {
      expect(setup(OutlineButton, true)).toMatchSnapshot();
      const wrapper = setup(OutlineButton, true, false, 'white');
      expect(wrapper.find(OutlineButton).prop('colour')).toBe('white');
      expect(wrapper).toHaveStyleRule('color', theme.colours.whiteDenim);
      expect(wrapper).toHaveStyleRule('background', theme.colours.whiteDenim);
    });

    it('renders correctly in loading state', () => {
      expect(setup(OutlineButton, false, true, 'black')).toMatchSnapshot();
      const wrapper = setup(OutlineButton, false, true, 'black');
      expect(wrapper.find(OutlineButton).prop('loading')).toBeTruthy();
    });
  });

  describe('Link Button', () => {
    it('renders correctly', () => {
      expect(setup(LinkButton)).toMatchSnapshot();
    });

    it('renders correctly in disabled state', () => {
      expect(setup(LinkButton, true)).toMatchSnapshot();
    });

    it('renders correctly in loading state', () => {
      expect(setup(LinkButton, false, true)).toMatchSnapshot();
    });
  });
});
