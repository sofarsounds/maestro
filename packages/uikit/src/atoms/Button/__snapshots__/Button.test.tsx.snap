// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Link Button renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Link Button renders correctly in disabled state 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Link Button renders correctly in loading state 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Primary Button renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Primary Button renders correctly in disabled state 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Primary Button renders correctly in loading state 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Secondary Button renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Secondary Button renders correctly in disabled state 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Secondary Button renders correctly in loading state 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Tertiary Button renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Tertiary Button renders correctly in disabled state 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Buttons Tertiary Button renders correctly in loading state 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
